docs/client-specifications/configuration.md

```diff
@@ -0,0 +1,147 @@
+  have established high-confidence default settings that cover the 90% use case, so that users don't have to spend
+  time figuring it out themselves.
+* Because there are different requirements for different environments (e.g. dev vs. prod), we provide a small handful of
+  pre-built configurations that users can choose from, out-of-the box.
+* Configurations must be specified as interfaces, so that alternate implementations can be added over time.
+* Users should always have a last-resort option of manually tuning the settings themselves, in case there are special
+  cases that our defaults are not quite right for.
+
+# Choosing a configuration
+
+The `SimpleCacheClient` constructor should accept a required argument whose type is our `Configuration` interface.  e.g.:
+
+```typescript
+import * as config from '@gomomento/sdk/config';
+
+const client = SimpleCacheClient({config: config.Dev.latest()});
+```
+
+# Pre-built Configuration Objects
+
+## Requirements for each pre-built configuration objects
+
+Each SDK must provide a namespace (via the idiomatic construct in the relevant programming language) that contains
+an enumerated list of pre-built configuration objects.
+
+All configuration objects must provide a `latest()` function that returns the latest "blessed" defaults from Momento.
+Documentation must clearly indicate that these values may change over time if Momento finds in the future that some
+different default values will provide a better or more performant user experience.
+
+All configuration objects must also provide a `default()` function that returns the original "blessed" defaults from
+Momento.  Documentation must clearly state that these values will not change over time; users can rely on them to
+be stable across new minor version releases of the SDKs.
+
+The `SimpleCacheClient` constructor should expose a required argument that is an instance of
+
+## Required pre-built configuration objects
+
+Each SDK must provide at least the following out-of-the-box configuration objects:
+
+```typescript
+import * as config from '@gomomento/sdk/config';
+
+/**
+ * Dev provides defaults suitable for a medium-to-high-latency dev environment.  Permissive timeouts, retries, potentially
+ * a higher number of connections, etc.
+ */
+const devConfig = config.Dev.latest();
+
+/**
+ * ProdLowLatency provides defaults suitable for a low-latency prod environment.  This is for the use case where the
+ * client is running on an application server in the same region as the Momento service.  It has more agressive timeouts
+ * and retry behavior, to ensure that cache unavailability doesn't force unacceptably high application latencies.
+ */
+const prodConfig = config.ProdLowLatency.latest();
+```
+
+## Configuration interface
+
+The configuration interface provides nested objects for retry strategy, middleware, and transport:
+
+```typescript
+interface Configuration {
+  retryStrategy: RetryStrategy,
+  middleware: Middleware,
```

Comment:
I feel like it might be more idiomatic to have this be `middlewares`, which would take in an array of middleware. I know that this impl would support multiple middlewares, its just that the impl of supporting those middlewares would be the customers job instead of ours.

Reply:
Yeah fair callout, I'll think about this.  We could certainly provide a CompositeMiddleware class so they wouldn't have to invent it.  But in this case I'm on the fence, you might be right.  Do you have any particular software in mind that you are using as your mental foundation for what is "idiomatic" for middleware?

Reply:
I think the one that comes to mind first is express js. The way its normally implemented is
```
app.use(cookieParser())
app.use(auth())
etc...
```

Reply:
I am also on the fence about it tho, just thought it was worth mentioning

Reply:
changed to an array